index.d.ts

```diff
@@ -0,0 +1,287 @@
+	fail(text?: string): Ora;
+
+	/**
+	 * Stop the spinner, change it to a yellow `⚠` and persist the current text, or `text` if provided.
+	 *
+	 * @param text will persist text if provided
+	 * @returns The spinner instance.
+	 */
+	warn(text?: string): Ora;
+
+	/**
+	 * Stop the spinner, change it to a blue `ℹ` and persist the current text, or `text` if provided.
+	 *
+	 * @param text will persist text if provided
+	 * @returns The spinner instance.
+	 */
+	info(text?: string): Ora;
+
+	/**
+	 * Stop the spinner and change the symbol or text.
+	 *
+	 * @param options
+	 * @returns The spinner instance.
+	 */
+	stopAndPersist(options?: PersistOptions): Ora;
+
+	/**
+	 * Clear the spinner.
+	 * @returns The spinner instance.
+	 */
+	clear(): Ora;
+
+	/**
+	 * Manually render a new frame.
+	 * @returns The spinner instance.
+	 */
+	render(): Ora;
+
+	/**
+	 * Get a new frame.
+	 * @returns The spinner instance.
+	 */
+	frame(): Ora;
+}
+
+export interface Options {
+	/**
+	 * Text to display after the spinner.
+	 */
+	text?: string;
+
+	/**
+	 * Name of one of the provided spinners. See [`example.js`](https://github.com/BendingBender/ora/blob/master/example.js)
+	 * in this repo if you want to test out different spinners. On Windows, it will always use the line spinner as
+	 * the Windows command-line doesn't have proper Unicode support.
+	 *
+	 * Or an object like:
+	 *
+	 * @example
+	 *
+	 * {
+	 *     interval: 80, // optional
+	 *     frames: ['-', '+', '-']
+	 * }
+	 *
+	 * @default 'dots'
+	 */
+	spinner?: SpinnerName | Spinner;
+
+	/**
+	 * Color of the spinner.
+	 * @default 'cyan'
+	 */
+	color?: Color;
+
+	/**
+	 * Set to `false` to stop Ora from hiding the cursor.
+	 * @default true
+	 */
+	hideCursor?: boolean;
+
+	/**
+	 * Indent the spinner with the given number of spaces.
+	 * @default 0
+	 */
+	indent?: number;
+
+	/**
+	 * Interval between each frame.
+	 *
+	 * Spinners provide their own recommended interval, so you don't really need to specify this.
+	 * @default Provided by the spinner or 100
+	 */
+	interval?: number;
+
+	/**
+	 * Stream to write the output.
+	 *
+	 * You could for example set this to `process.stdout` instead.
+	 * @default process.stderr
+	 */
+	stream?: NodeJS.WritableStream;
+
+	/**
+	 * Force enable/disable the spinner. If not specified, the spinner will be enabled
+	 * if the `stream` is being run inside a TTY context (not spawned or piped) and/or not in a CI environment.
+	 *
+	 * Note that `{isEnabled: false}` doesn't mean it won't output anything. It just means it won't output the spinner,
+	 * colors, and other ansi escape codes. It will still log text.
+	 */
+	isEnabled?: boolean;
+}
+
+export interface PersistOptions {
+	/**
+	 * Symbol to replace the spinner with.
+	 * @default ' '
+	 */
+	symbol?: string;
+
+	/**
+	 * Text to be persisted.
+	 * @default Current text
+	 */
+	text?: string;
+}
+
+export interface Spinner {
+	interval?: number;
+	frames: string[];
+}
+
+export type SpinnerName =
```

Comment:
This should be defined in https://github.com/sindresorhus/cli-spinners and imported from there instead.

Reply:
Ok, will do the types for cli-spinners first.